ssl/statem/statem_clnt.c

```diff
@@ -2508,11 +2508,8 @@ MSG_PROCESS_RETURN tls_process_new_session_ticket(SSL *s, PACKET *pkt)
         s->session = new_sess;
     }
 
-    /*
-     * Technically the cast to long here is not guaranteed by the C standard -
-     * but we use it elsewhere, so this should be ok.
-     */
-    s->session->time = (long)time(NULL);
+    s->session->time = time(NULL);
```

Comment:
Locking?

Reply:
No worse than as originally written. Yhis is in the client, where sharing sessions is less common.

Reply:
This whole function (`tls_process_new_session_ticket()`) is creating a new (or duplicate) SSL_SESSION. The session structure is heavily modified (without locking) after this line.